### ISSUE TITLE
In WP-admin Categories are not shown in Posts

### ISSUE THREAD

**Issue by kspilarski** [Automattic/wp-calypso#64552]
### Quick summary

In WP-admin, Categories are not shown in Posts. Bulk deletion and bulk editing are affected.  

### Steps to reproduce

1. Assign posts to any category. 
2. Go to WP-admin  > Posts 
3. Under Bulk Actions, select Edit then Apply
4. The posts that are assigned to a category don't show that they are already in that category. 

### What you expected to happen

The categories the posts are assigned to are displayed. 

### What actually happened

The posts assigned to categories don't show which category they belong to. 

### Context

Customer report: 5289034-zd and self-testing.

In my test site, these three posts are in Desserts and Drinks categories: 

 
<img width="1089" alt="Screen Shot 2022-06-11 at 4 19 53 PM" src="https://user-images.githubusercontent.com/36863633/173192177-3b3f28c9-4a3e-4530-bbf9-bbb567838524.png">

In WP-admin > Posts > Bulk edit posts don't show that these posts are in Desserts and Drinks categories. I can't tell which category they belong to. 

<img width="1188" alt="Screen Shot 2022-06-11 at 4 21 38 PM" src="https://user-images.githubusercontent.com/36863633/173192008-8ec12c20-7c3d-42a8-9f39-e1c0e67b5ec8.png">

In WP-admin Posts, even I filter the posts by category (example: desserts category), desserts category box is not ticked. I am unable to remove posts from these categories: 

<img width="713" alt="Screen Shot 2022-06-11 at 4 33 07 PM" src="https://user-images.githubusercontent.com/36863633/173192247-f3c64d75-f2c3-4d62-9540-7168cf57f58d.png">

### Simple, Atomic or both?

Simple, Atomic

### Browser, operating system and other notes

Firefox 101, macOS 12.4 

### Reproducibility

Consistent

### Severity

All

### Available workarounds?

Yes, easy to implement

### Workaround details

To delete the category in all posts, go to Posts > Categories and delete the category completely. 

To delete certain posts in the category, open the post one by one and remove the category from that post.

**Comment by sstabrizi** [comment]
Another report in 35810422-hc. 

I'd add one more related issue. When looking under Posts > (Click Post Title) > Categories, the same behavior happens as above. If you select a category, as soon as Update is clicked, the category is removed from the editor, but the post still shows in the category feed. This also means it's not possible to uncategorize a post.

I tried disabling all user-installed plugins to see if was plugin related, and it did not do anything. This fits with the above description that this was happening on atomic and simple sites when it was reported.

**Comment by renata-franco** [comment]
📌 HOUSEKEEPING
- Labels ✅ 
- Priority ✅ 
- Replicable on Core - Yes

📌 SCRUBBING
- Tested on P2 ✅ 
- Tested on Simple ✅ 
- Tested on AT ✅ 
- Tested on Self-hosted ✅ 

📌 FINDINGS/SCREENSHOTS/VIDEO

I was able to replicate the categories issue on Simple, AT, and Self-Hosted. The selected posts to edit in bulk do not appear in the categories they already have assigned, none of them, actually.

Regarding the second comment about this happening also with the pages that @sstabrizi left I am not sure what they meant, as the Pages do not have Categories. 

As for a workaround, there is no other way to edit the Categories in bulk natively. 

**Simple Site**

![image](https://user-images.githubusercontent.com/81378719/174111561-d906615d-d9f9-44ab-8e8e-eb9edf691ad0.png)

**AT Site**

![image](https://user-images.githubusercontent.com/81378719/174111962-f263988f-b051-45e8-9b00-718805561360.png)

**Self-Hosted Site**

![image](https://user-images.githubusercontent.com/81378719/174112191-b9e3fa6a-53dd-4419-9026-ec91b51caccf.png)





📌 ACTIONS

- Marked as Triaged for Quality Squad review.

**Comment by sstabrizi** [comment]
> Regarding the second comment about this happening also with the pages that @sstabrizi left I am not sure what they meant, as the Pages do not have Categories.

Just updated my comment above. I meant Posts > (Click Post Title) > Categories (not pages). What I was trying to say was it's not just the bulk or quick edit that doesn't work. It's also not working when you categorize from the post editor. Here are some screenshots:

**Before clicking Update**

<img width="364" alt="Markup 2022-06-16 at 11 09 51" src="https://user-images.githubusercontent.com/25855381/174127957-6855dffd-5b62-49e7-aba5-28cab4037e64.png">

**After clicking Update**

<img width="361" alt="Markup 2022-06-16 at 11 10 13" src="https://user-images.githubusercontent.com/25855381/174127944-ab869c36-91c8-4b2b-a5db-3b361df94eb2.png">

**Comment by renata-franco** [comment]
Oh, thanks @sstabrizi. Can you please create a new bug for that other one?

**Comment by sstabrizi** [comment]
Sorry, now I'm confused. This seems to be the same issue, and the behavior is just showing in two different places. Surely the same issue is at the root of both of these, no?

**Comment by renata-franco** [comment]
I checked with one of my colleagues, @sstabrizi and we will let the developers decide if this should be a different bug as one is happening in Calypso and the other one in WP-admin. In the meantime, thank you so much for sharing the screenshots!

**Comment by kuropixel** [comment]
Adding the ticket for previously reported 35810422-hc:
 5301111-zen
 
 They'd like to be notified when the bug is fixed.

**Comment by cometgrrl** [comment]
> I checked with one of my colleagues, @sstabrizi and we will let the developers decide if this should be a different bug as one is happening in Calypso and the other one in WP-admin. In the meantime, thank you so much for sharing the screenshots!

We should start with filing a core bug as getting that fixed may solve both issues.

**Comment by mrfoxtalbot** [comment]
I can replicate the initial/main issue (let's call it "Issue A") but it took me a while to understand it. Another more descriptive title might be "Bulk editing multiple posts will not list all the relevant categories". I would strongly suggest including a short video recording whenever possible to make it easier to understand.

https://user-images.githubusercontent.com/4452464/177126900-27b7b32a-c096-41ae-a0ca-1a9ff5416f70.mov

@sstabrizi I am not able to replicate what you describe (let's call it "Issue B"). Could you please confirm that you are still able to replicate this on a wpcom simple site and/or on a self-hosted site with no plugins and Twenty Twenty-Two active? There is a short video illustrating what I am seeing on my end:

https://user-images.githubusercontent.com/4452464/177126749-4d5f7ea4-51d6-4e27-a441-9371ba048fd6.mp4

Thank you for adding the `Core Fix Needed` tag @cometgrrl! 

I found that this is being discussed in [this trac issue](https://core.trac.wordpress.org/ticket/11302). There has been a patch in place ready to ship for about 18 months at this point but I am not sure what is causing the delay to deploy. Are you familiar with this issue @priethor, @annezazu?

**Comment by kcswong** [comment]
Encountered this in 36697186-hc, tested in both WP Admin and Calypso.
Follow-up ticket: 5566734-zen

**Comment by kspilarski** [comment]
Same case in 6233188-zd.

**Comment by jp-imagines** [comment]
Another report in 6243650-zd-woothemes.

**Comment by caitlindawn** [comment]
Another case in 7089123-zd.

**Comment by cuemarie** [comment]
https://core.trac.wordpress.org/ticket/11302 was closed 9 days ago with a fix that pre-fills post categories either with a checked checkbox, or a line indicating an indeterminate status, when bulk editing posts - that hasn't reached my self-hosted or dotcom sites yet, but it appears that is on it's way.

📌 ACTIONS
- Due to the age of this issue, I've downgraded the priority from `High` to `Normal` to better reflect its status as waiting on that Core update.

**Comment by mrfoxtalbot** [comment]
I retested this and I can confirm it is fixed (All relevant categories are listed when editing posts in bulk).

![Screenshot 2024-09-02 at 14 11 33](https://github.com/user-attachments/assets/75f3def5-0c23-4b85-adf8-a80f8ad2e30b)
